### ISSUE TITLE
Parse `import.meta`

### ISSUE THREAD

**Issue by ravinggenius** [eslint/eslint#10271]
**Please describe what the rule should do:**

Parse ES module metadata: `import.meta`.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[X] Other (please specify:)

`import.meta` is a new addition to ES modules. Node v10 added support for `import.meta` when run with `--experimental-modules`.

**Provide 2-3 code examples that this rule will warn about:**

These are examples of correct usage. Currently eslint (`v4.19.1`) reports an incorrect parse error at the `import` keyword.

```js
// import.meta is a regular javascript object, so property access (read and write)
// and destructuring should work
const { url } = import.meta;
```

```js
// import.meta is file-local and by default only has one property, url
// creating new properties is allowed
import.meta.answer = 42;
```

```js
const helper = ({ url }) => url.replace(/^file:\/\//, '');
helper(import.meta);
```

**Why should this rule be included in ESLint (instead of a plugin)?**

`import.meta` is a part of the language. It is likely to be used, as `import.meta.url` is the only way to compute `__filename` in ES modules. (Node used to provide `__filename` as a pre-defined variable.)

**Comment by not-an-aardvark** [comment]
Thanks for the proposal. `import.meta` is a stage 3 feature, and as described in our [README](https://github.com/eslint/eslint/tree/b32d1f48cecc1eecd1352a978d7b393ad047337c#what-about-experimental-features) we don't support experimental syntax. I would recommend using a custom parser such as [`babel-eslint`](https://github.com/babel/babel-eslint) instead.

**Comment by Aladdin-ADD** [comment]
Closing because this is working as intended.
